lib/chef/resource.rb

```diff
@@ -960,6 +964,11 @@ def method_missing(method_symbol, *args, &block)
       end
     end
 
+    def self.provides(name, *args, &block)
+      super
```

Comment:
Should be `super.tap { Chef::DSL::Resources.add_resource_dsl(name) }` or similar to avoid blocking the return value. Most of my resources use `.provides()` as an accessor too.